developers_chamber/jira_utils.py

```diff
@@ -0,0 +1,106 @@
+import math
+
+import click
+
+import unidecode
+
+import webbrowser
+
+from requests.exceptions import ConnectionError
+
+from jira import JIRA, JIRAError
+
+from .git_utils import get_current_issue_key
+
+
+def clean_issue_key(issue_key=None, project_key=None):
+    if not issue_key:
+        issue_key = get_current_issue_key()
+
+    if not issue_key:
+        raise click.UsageError('Please set task key')
+
+    if project_key and not issue_key.startswith('{}-'.format(project_key)):
+        return '{}-{}'.format(project_key, issue_key)
+    else:
+        return issue_key
+
+
+def get_jira_client(url, username, api_key):
+    try:
+        client = JIRA(url, basic_auth=(username, api_key))
+        return client
+    except ConnectionError:
+        raise click.BadParameter('Invalid Jira URL or connection problem')
+
+
+def get_current_user_issues(url, username, api_key, project_key, jql):
+    issues = get_jira_client(url, username, api_key).search_issues(jql.format(
+        project_key=project_key
+    ))
+    return '\n'.join(
+        ['    - {}: {} ({})'.format(issue.key, issue.fields.summary, issue.permalink()) for issue in issues]
+    )
+
+
+def get_branch_name(url, username, api_key, issue_key, project_key=None):
+    issue_key = clean_issue_key(issue_key, project_key)
+
+    try:
+        issue = get_jira_client(url, username, api_key).issue(issue_key)
+        return '{}_{}'.format(issue.key, unidecode.unidecode(issue.fields.summary).replace(' ', '-').lower())[:40]
```

Comment:
Strictly speaking, this should go to the `else` branch (applies to other try/except constructs too)

Reply:
No is not necessary because jira error can raise only the first line